### PR TITLE
cleaned up if to make assignment obvious

### PR DESCRIPTION
**Pull request by arranbartish**
Very minor readability change. 

The CollectionUtils isEmpty null safe and gives us the same result as the two if statements. 

I then found that the branches attribute was getting assigned at a different level as the empty check so I replaces the if to make it obvious that this.branches was always going to be assigned.

### CODE REVIEW

**Comment by KostyaSha** [comment]
:+1: looks valid for me

File: src/main/java/hudson/plugins/git/GitSCM.java [hunk 1]
```diff
@@ -161,13 +164,7 @@ public GitSCM(
...
 
         // moved from createBranches
-        if (branches == null) {
-            branches = new ArrayList<BranchSpec>();
-        }
-        if (branches.isEmpty()) {
-            branches.add(new BranchSpec("*/master"));
-        }
-        this.branches = branches;
+        this.branches = isEmpty(branches) ? newArrayList(new BranchSpec("*/master")) : branches;
```
Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitSCM.java]: shouldn't newArrayList be in two words?

Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitSCM.java]: nevermind, so many core libs with features

**Comment by KostyaSha** [comment]
:+1: looks ok for me

**Comment by MarkEWaite** [comment]
Rebased into the stream that will be git plugin 2.4

**Pull request closed without merging**